platform/FEATURE_EXPERIMENTAL_API/FEATURE_PSA/TARGET_MBED_PSA_SRV/mbedtls/psa_crypto.c

```diff
@@ -5946,8 +6491,6 @@ static psa_status_t psa_generate_key_internal(
             return( PSA_ERROR_NOT_SUPPORTED );
         if( grp_id == MBEDTLS_ECP_DP_NONE || curve_info == NULL )
             return( PSA_ERROR_NOT_SUPPORTED );
-        if( curve_info->bit_size != bits )
-            return( PSA_ERROR_INVALID_ARGUMENT );
         mbedtls_ecp_keypair_init( &ecp );
         ret = mbedtls_ecp_gen_key( grp_id, &ecp,
                                    mbedtls_ctr_drbg_random,
```

Comment:
For Mbed PSA targets (e.g. K64F), we get undefined symbol error for `mbedtls_ecp_gen_key()`.

This call here is inside `#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR)`, which is enabled set the newly added `connectivity/mbedtls/include/mbedtls/config_psa.h` if `MBEDTLS_ECP_C` is defined.

The definition `mbedtls_ecp_gen_key()` in `connectivity/mbedtls/source/rsa.c` requires`MBEDTLS_GENPRIME`. But our script unsets it: https://github.com/ARMmbed/mbed-os/blob/47b19eb157e64c35d7164431c9b469979742b4d5/connectivity/mbedtls/tools/importer/adjust-config.sh#L85 The comment from five years ago says https://github.com/ARMmbed/mbed-os/blob/47b19eb157e64c35d7164431c9b469979742b4d5/connectivity/mbedtls/tools/importer/adjust-config.sh#L54

I wonder if we should enable `MBEDTLS_GENPRIME` globally (if current targets support it), or enable it for Mbed PSA targets, or disable `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR` to avoid the call?

Reply:
We'd like to avoid enabling GENPRIME_C if possible, but still be able to use PSA to import RSA keys or sign with them (at least verify would be good). I'll dig to see if this is a permitted configuration.